Kitodo/src/main/java/org/kitodo/services/file/FileService.java

```diff
@@ -564,48 +569,77 @@ public URI getProcessBaseUriForExistingProcess(Process process) {
      *            the process to get the sublocation for.
      * @param processSubType
      *            The subType.
-     * @param id
+     * @param folderName
      *            the id of the single object (e.g. image) if null, the root
      *            folder of the sublocation is returned
      * @return The URI of the requested location
      */
-    public URI getProcessSubTypeURI(Process process, ProcessSubType processSubType, String id) {
+    public URI getProcessSubTypeURI(Process process, ProcessSubType processSubType, String folderName) {
 
         URI processDataDirectory = serviceManager.getProcessService().getProcessDataDirectory(process);
 
-        if (id == null) {
-            id = "";
+        if (folderName == null) {
+            folderName = "";
+        }
+
+        return URI.create(processDataDirectory + getProcessSubType(process, processSubType, folderName));
+    }
+
+    /**
+     * Unmap process specific part of URI e.g 3/images. Lack of this method was
+     * causing error of double uri creation e.g 3/images/3/images/scans_tif
+     * 
+     * @param uriList
+     *            list of URIs for unmap
+     * @param process
+     *            object
+     * @param processSubType
+     *            object
+     * @param folderName
+     *            as String
+     * @return List of unmapped URIs
+     */
+    public ArrayList<URI> unmapProcessSpecificPartOfUri(ArrayList<URI> uriList, Process process,
+            ProcessSubType processSubType, String folderName) {
+        ArrayList<URI> unmapedURI = new ArrayList<>();
+        for (URI uri : uriList) {
+            String uriString = uri.toString();
+            String processSpecificPartOfUri = getProcessSubType(process, processSubType, folderName);
+            if (uriString.contains(getProcessSubType(process, processSubType, folderName))) {
+                String[] split = uriString.split(processSpecificPartOfUri);
+                String shortUri = split[1];
+                unmapedURI.add(URI.create(shortUri));
+            }
+            unmapedURI.add(uri);
         }
+        return unmapedURI;
+    }
 
+    private String getProcessSubType(Process process, ProcessSubType processSubType, String folderName) {
```

Comment:
Should be done by a rebase against master as `resourceName` is already used (change was in PR #850).